workflows/prognostic_c48_run/tests/test_fv3fit_transformer.py

```diff
@@ -71,42 +89,41 @@ def add_one_to_temperature():
     regression_state(out, regtest)
 
 
-def test_tendency_or_state_multimodel_adapter(state):
-    nz = 63
-    outputs = {
-        "Q1": np.full(nz, 1 / 86400),
-        "tendency_of_air_temperature_due_to_nudging": np.full(nz, 1 / 86400),
-        "implied_surface_precipitation_rate": 10 / 86400,
-    }
-    predictor_one = fv3fit.testing.ConstantOutputPredictor(
-        input_variables=["air_temperature", "specific_humidity"],
-        output_variables=["Q1", "implied_surface_precipitation_rate"],
-    )
-    predictor_two = fv3fit.testing.ConstantOutputPredictor(
-        input_variables=["air_temperature", "land_sea_mask"],
-        output_variables=["tendency_of_air_temperature_due_to_nudging"],
-    )
-    predictor_one.set_outputs(**outputs)
-    predictor_two.set_outputs(**outputs)
-    multi_model_predictor = TendencyOrStateMultiModelAdapter(
-        [predictor_one, predictor_two],
+def test_prediction_inverter():
```

Comment:
This test is still complex because it's testing things like timesteps via an integration test that could be covered by smaller unit tests. It also uses hardcoded names/constants unecessarily.


Here's how I would write the tests to follow the abstractions you've defined. First, the appliers are trivial to test individually:
```
def test_MLOutputApplier_tendnecy():
    x = ...some dataset with "a" and "tendency"
    applier = MLOutputApplier("a", "tendency", timestep=1)
    assert applier.apply(x) - x.a == x.tendency # won't work, but you know what I mean

def test_ML_OutputApplier_state():
     # super easy to test
```
With this unit test done, I now have coverage of input to output name translation, timesteps, tendencies, etc. I would intentionally avoid these concepts when testing `PredictionInverter`. Assuming you refactor the `.invert` method as I suggest, then there is almost nothing to "unit" test" any longer...the `Inverter` mostly just wires some other objects together without changing any data. mypy and other integration tests should cover this wiring logic well enough. Since it doesn't do much, I'd put a high probability on the `Inverter` changing in some future refactor, and it would be nice not to have this test prevent that from happening. OTOH, "MLOutputApplier" will likely be a much more resilient API.

tl;dr: add some unit tests of `MLOuptutApplier` and delete this test.

Reply:
Not totally following here. It is not possible to apply the tendencies one-by-one to the state, because of the desire to limit the **total** tendency. So we need to sum the tendencies for each variable, then pass these summed tendencies to the limiter, and then apply to the state. This test exists primarily to ensure that the tendencies are summed correctly.